src/accessibility/describe.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * @module Environment
+ * @submodule Environment
+ * @for p5
+ * @requires core
+ */
+
+import p5 from '../core/main';
+
+/**
+ * Creates a screen-reader accessible description for the canvas in the DOM.
+ * The first parameter should be a string with a description of the canvas.
+ * The second parameter is optional. If specified, it determines how the
+ * description is displayed.
+ *
+ * <code class="language-javascript">describe(text, LABEL)</code> displays
+ * the description to all users as a <a
+ * href="https://en.wikipedia.org/wiki/Museum_label" target="_blank">
+ * tombstone or exhibit label/caption</a> in a
+ * <code class="language-javascript">&lt;div class="p5Label"&gt;&lt;/div&gt;</code>
+ * adjacent to the canvas. You can style it as you wish in your CSS.
+ *
+ * <code class="language-javascript">describe(text, FALLBACK)</code> makes the
+ * description accessible to screen-reader users only, in
+ * <a href="https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Hit_regions_and_accessibility" target="_blank">
+ * a sub DOM inside the canvas element</a>. If a second parameter is not
+ * specified, by default, the description will only be available to
+ * screen-reader users.
+ *
+ * @method describe
+ * @param  {String} text      description of the canvas
+ * @param  {Constant} [display] either LABEL or FALLBACK (Optional)
+ *
+ * @example
+ * <div>
+ * <code>
+ * describe('pink square with red heart in the bottom right corner', LABEL);
+ * background('pink');
+ * fill('red');
+ * noStroke();
+ * ellipse(66.6, 66.6, 20, 20);
+ * ellipse(83.2, 66.6, 20, 20);
+ * triangle(91.2, 72.6, 75, 95, 58.6, 72.6);
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * let x = 0;
+ * function draw() {
+ *   if (x > 100) {
+ *     x = 0;
+ *   }
+ *   background(220);
+ *   fill(0, 255, 0);
+ *   ellipse(x, 50, 40, 40);
+ *   x = x + 0.1;
+ *   describe('green circle at x pos ' + round(x) + ' moving to the right');
+ * }
+ * </code>
+ * </div>
+ *
+ */
+p5.prototype.describe = function(t, d) {
+  p5._validateParameters('describe', arguments);
+  const cnvId = this.canvas.id;
+  //Creates a sub DOM inside of the canvas element and populates
+  //it with description text.
+  t = this._descriptionText(t);
+  if (document.getElementById(cnvId + '_Description') === null) {
+    document.getElementById(cnvId).innerHTML =
+      '<div id="' +
+      cnvId +
+      '_Description" role="region" aria-label="Canvas Description"><p id="' +
+      cnvId +
+      '_dsc"></p></div>';
+  } else if (document.getElementById(cnvId + '_dsc') === null) {
+    document
+      .getElementById(cnvId + '_eDsc')
+      .insertAdjacentHTML('beforebegin', '<p id="' + cnvId + '_dsc"></p>');
+  }
+  if (document.getElementById(cnvId + '_dsc').innerHTML !== t) {
```

Comment:
It seems safe to just assign without checking, but does this keep it from notifying the screen reader that the value has been updated? (asking for clarity, for me) :)